Production/src/BaseTupleProducer.cc

```diff
@@ -161,10 +161,14 @@ void BaseTupleProducer::InitializeCandidateCollections(analysis::EventEnergyScal
     for(const auto& muon : *pat_muons)
         muons.push_back(MuonCandidate(muon, Isolation(muon)));
 
+
     taus.clear();
     for(const auto& tau : *pat_taus) {
         TauCandidate tauCandidate(tau, Isolation(tau));
         if(tauEnergyScales.count(energyScale)) {
+            const analysis::gen_truth::MatchResult result =
+                    analysis::gen_truth::LeptonGenMatch(tauCandidate.GetMomentum(), *genParticles);
+            if(result.first != analysis::GenMatch::Tau) continue;
```

Comment:
continue will skip not-matching taus, which should not be the case. Remove continue and put next 4 lines inside the if.